src/Appwrite/Utopia/Response/Model/Message.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace Appwrite\Utopia\Response\Model;
+
+use Appwrite\Utopia\Response;
+use Appwrite\Utopia\Response\Model;
+use Utopia\Database\DateTime;
+
+class Message extends Any
+{
+    public function __construct()
+    {
+        $this
+            ->addRule('$id', [
+                'type' => self::TYPE_STRING,
+                'description' => 'Message ID.',
+                'default' => '',
+                'example' => '5e5ea5c16897e',
+            ])
+            ->addRule('providerId', [
+                'type' => self::TYPE_STRING,
+                'description' => 'Provider ID for the message.',
+                'default' => '',
+                'example' => '5e5ea5c16897e',
+            ])
+            ->addRule('to', [
+                'type' => self::TYPE_STRING,
+                'description' => 'Message recipients.',
+                'default' => '',
+                'array' => true,
+                'example' => ['user-1'],
+            ])
+            ->addRule('deliveryTime', [
+                'type' => self::TYPE_DATETIME,
+                'description' => 'The scheduled time for message.',
+                'required' => false,
+                'default' => DateTime::now(),
+                'example' => self::TYPE_DATETIME_EXAMPLE,
+            ])
+            ->addRule('deliveredAt', [
+                'type' => self::TYPE_DATETIME,
+                'description' => 'The time when the message was delivered.',
+                'required' => false,
+                'default' => '',
+                'example' => self::TYPE_DATETIME_EXAMPLE,
+            ])
+            ->addRule('deliveryErrors', [
+                'type' => self::TYPE_STRING,
+                'description' => 'Delivery errors if any.',
+                'required' => false,
+                'default' => '',
+                'array' => true,
+                'example' => ['Failed to send message to target 5e5ea5c16897e: Credentials not valid.'],
+            ])
+            ->addRule('deliveredTo', [
+                'type' => self::TYPE_INTEGER,
+                'description' => 'Number of recipients the message was delivered to.',
+                'default' => 0,
+                'example' => 1,
+            ])
+            ->addRule('data', [
```

Comment:
should this be data / payload / body? how does this compare to other similar fields we have?

Reply:
Actually I think this should be removed so we have the same as documents where the the attributes are added at the top level :thinking:

Reply:
@abnegate, I think @stnguyen90 , might need ``data`` attribute in the message response

Reply:
@stnguyen90 Does it need to be inside `data` or can it be treated like a Document, where the attributes inside `data` are hoisted to the top level?

Reply:
Either is fine.